### PR TITLE
feat: add hapify meta for search and unique index configuration

### PR DESCRIPTION
**Pull request by maxmousse**

### CODE REVIEW

File: libs/hapify/templates/prisma/hapify/prisma/schemas/generated.prisma.hpf [hunk 1]
```diff
@@ -234,11 +237,71 @@ function primary(f) {
...
+                }
+            });
+            return indexes;
+        }, {})
+
+    // construct @@unique decorators from field names
+    return Object.values(indexes).map(index => `@@unique([ ${index.map(index => index.names.camel).join(',') } ])`).join('\n');
+}
+
+function researchIndexes(model) {
```
Comment (EdouardDem) [inline libs/hapify/templates/prisma/hapify/prisma/schemas/generated.prisma.hpf]: Did you mean `searchIndexes` ?

**Review by EdouardDem — COMMENTED** [review]
You should replace `Research` by `Search` everywhere

File: libs/hapify/templates/prisma/hapify/prisma/schemas/generated.prisma.hpf (hunk 1 above)
Comment (maxmousse) [reply libs/hapify/templates/prisma/hapify/prisma/schemas/generated.prisma.hpf]: I changed the 'research' occurences into 'search'

**Comment by maxmousse** [comment]
@floross @EdouardDem There is a TODO comment in the hapify validator where I would like to have your opinion. Also, I was wondering if we should the names used into the hapify meta to name the indexes. It can be useful for composite uniqueness indexes (see https://www.prisma.io/docs/concepts/components/prisma-schema/names-in-underlying-database#related-naming-indexes-and-primary-keys-for-prisma-client). We could also add a way to handle composite primary keys.

**Review by floross — APPROVED** [review]

**Pull request merged**